examples/gc/sha256.cc

```diff
@@ -0,0 +1,319 @@
+//       }
+//     }
+//   }
+//   std::cout << endl;
+// }
+
+// template <typename T>
+// void PlainExecutor<T>::SetupInputs(
+//     vector<uint8_t> inputs_bi) {  // Span方便指针的使用
+//   // YACL_ENFORCE(inputs.size() == circ_->niv);
+//   for (auto input : inputs_bi) {
+//     wires_.append(
+//         std::bitset<8>(input).begin(),
+//         std::bitset<8>(input).end());  // 直接转换为二进制 输入线路在前128位
+//   }
+//   wires_.resize(circ_->nw);
+// }
+
+// #include <bits/stdc++.h>
+
+// std::vector<uint8_t> stringToBinary(const std::string& str) {
+//   std::vector<uint8_t> binaryData(str.begin(), str.end());
+//   return binaryData;
+// }
+
+// int main() {
+//   std::string str = "Hello";
+//   std::vector<uint8_t> binaryData = stringToBinary(str);
+
+//   // 打印二进制数据
+//   for (uint8_t byte : binaryData) {
+//     std::std::cout << std::bitset<8>(byte) << " ";
+//   }
+//   std::std::cout << std::endl;
+
+//   return 0;
+// }
+
+// void get_sha256_initial_hash_values(uint32_t hash[8]) {
+//   // SHA256 的初始哈希值（每个值为 32 位）
+//   hash[0] = 0x6a09e667;
+//   hash[1] = 0xbb67ae85;
+//   hash[2] = 0x3c6ef372;
+//   hash[3] = 0xa54ff53a;
+//   hash[4] = 0x510e527f;
+//   hash[5] = 0x9b05688c;
+//   hash[6] = 0x1f83d9ab;
+//   hash[7] = 0x5be0cd19;
+// }
+int main() {
+  // const char *data = "Hello, OpenSSL!";
+  // size_t data_len = strlen(data);
+
+  // // 进行 SHA256 预处理
+  // uint8_t *padded_data;
+  // size_t padded_len;
+  // sha256_preprocess((uint8_t *)data, data_len, &padded_data, &padded_len);
+  // std::cout << "预处理：";
+  // for (int i = 0; i < 64; i++) {
+  //   std::cout << std::hex << static_cast<int>(padded_data[i]);
+  // }
+  // std::cout << endl;
+
+  // vector<uint8_t> data_vec(padded_data, padded_data + 64);
+
+  std::string input;
+  std::cout << "请输入:";
+  std::cin >> input;
+  std::vector<uint8_t> message(input.begin(), input.end());
+
+  // 预处理消息
+  std::vector<uint8_t> data_vec = preprocess(message);
+
+  // std::vector<uint32_t> hash_vec = {0x6a09e667, 0xbb67ae85, 0x3c6ef372,
+  //                                   0xa54ff53a, 0x510e527f, 0x9b05688c,
+  //                                   0x1f83d9ab, 0x5be0cd19};
+
+  /*
+  Initial hash value, reference:
+  https://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.180-4.pdf
+  Section 5.3
+  */
+  std::vector<uint8_t> byte_hash_vec = {
+      0x6a, 0x09, 0xe6, 0x67, 0xbb, 0x67, 0xae, 0x85, 0x3c, 0x6e, 0xf3,
+      0x72, 0xa5, 0x4f, 0xf5, 0x3a, 0x51, 0x0e, 0x52, 0x7f, 0x9b, 0x05,
+      0x68, 0x8c, 0x1f, 0x83, 0xd9, 0xab, 0x5b, 0xe0, 0xcd, 0x19};
+
+  // 遍历 hash_vec 并将每个 uint32_t 分解为 4 个字节存储到 byte_hash_vec 中
+  // for (uint32_t value : hash_vec) {
+  //   byte_hash_vec.push_back(value >> 24 & 0xFF);
+  //   byte_hash_vec.push_back((value >> 16) & 0xFF);
+  //   byte_hash_vec.push_back((value >> 8) & 0xFF);
+  //   byte_hash_vec.push_back((value) & 0xFF);
+  // }
+  for (auto &elem : data_vec) {
+    elem = reverse_bits(elem);
```

Comment:
can avoid reverse_bits by using `SetupInputs` api

Reply:
For data blocks: 
       a1         ||          a0
0101 0010  ||  1110 0101

We should feed:
       a0        ||         a1
1110 0101   || 0101 0010

So , just reverse bytes are sufficient.